tests/integration.rs

```diff
@@ -120,6 +120,35 @@ fn invert() -> Result<()> {
   assert_output_eq(&["resize:1:1", "all"], "1")
 }
 
+#[test]
+fn save() -> Result<()> {
+  assert_output_eq(
+    &["resize:1:2", "top", "save:output.png"],
+    "1
+     0",
+  )?;
+
+  let image = image::open("output.png")?;
+  let image = image.as_rgb8().unwrap();
+
+  assert_eq!(image.width(), 2);
+  assert_eq!(image.height(), 1);
+  assert_eq!(image.to_vec(), &[255, 255, 255, 0, 0, 0]);
```

Comment:
Wait, shouldn't this be:

```suggestion
  assert_eq!(image.width(), 1);
  assert_eq!(image.height(), 2);
  assert_eq!(image.to_vec(), &[255, 255, 255, 0, 0, 0]);
```

Reply:
I thought so as well, but I suppose its because `image` uses row-major order instead of nalgebra's column-major order?

Reply:
I don't think that would affect the width and the height, which should match the width and height of the image that's printed to the terminal.

Reply:
Since we transpose the matrix, I think we missed flipping (rows, cols) to (cols, rows) in `state::image`. My latest commit fixes this and fixes the test.